common/thrift_router.h

```diff
@@ -377,9 +387,30 @@ class ThriftRouter {
         const Role role,
         const unsigned rotation_counter,
         const std::string& segment,
-        const int shrink_target) {
+        const int shrink_target,
+        const std::string& specific_az) {
       std::vector<const Host*> v;
-      if (role == Role::ANY && !FLAGS_always_prefer_local_host) {
+      if (specific_az != "") {
+        v.reserve(host_info.size());
+        std::vector<const Host*> v_s;
+        // if az is specified by caller, return hosts belong only to that az.
+        for (const auto& hi : host_info) {
+          if (hi.first->az.compare(specific_az) == 0) {
+            if ((role == Role::ANY && hi.second == Role::MASTER) || role == hi.second) {
```

Comment:
if `role == Role::SLAVE`, this means we'll push slave nodes to `v`, and master to `v_s`, and select both in the final results, is that intended?